static/src/js/components/GranuleResults/__tests__/GranuleResultsDataLinksButton.test.jsx

```diff
@@ -27,63 +34,53 @@ function setup(overrideProps) {
     ],
     id: 'G123456789-TEST',
     s3Links: [],
-    onMetricsDataAccess: jest.fn(),
-    ...overrideProps
+    onMetricsDataAccess: jest.fn()
   }
+})
 
-  const enzymeWrapper = shallow(<GranuleResultsDataLinksButton {...props} />)
-
-  return {
-    enzymeWrapper,
-    props
+const setupCustomDataLinksToggle = setupTest({
+  Component: CustomDataLinksToggle,
+  defaultProps: {
+    id: 'G-123456789',
+    onClick: jest.fn()
   }
-}
-
-beforeEach(() => {
-  jest.clearAllMocks()
-
-  Object.assign(navigator, {
-    clipboard: {}
-  })
 })
 
 describe('GranuleResultsDataLinksButton component', () => {
-  test('renders itself correctly', () => {
-    const { enzymeWrapper } = setup()
-
-    expect(enzymeWrapper.type()).toBe(Button)
-  })
-
   describe('with no granule links', () => {
     test('renders a disabled download button', () => {
-      const { enzymeWrapper } = setup({
-        dataLinks: []
+      setup({
+        overrideProps: {
+          dataLinks: []
+        }
       })
 
-      expect(enzymeWrapper.find(Button).props().disabled).toBe(true)
-      expect(enzymeWrapper.type()).toBe(Button)
+      expect(screen.getByRole('button')).toBeDisabled()
     })
 
     test('prevents default when clicked', () => {
-      const preventDefaultMock = jest.fn()
-      const { enzymeWrapper } = setup({
-        dataLinks: []
-      })
+      const stopPropagationMock = jest.fn()
+
+      setup()
 
-      enzymeWrapper.find(Button).simulate('click', { preventDefault: preventDefaultMock })
-      expect(preventDefaultMock).toHaveBeenCalledTimes(1)
+      const button = screen.getByRole('button')
+
+      const clickEvent = createEvent.click(button)
+      clickEvent.stopPropagation = stopPropagationMock
+
+      fireEvent(button, clickEvent)
```

Comment:
we have to use fireEvent because we are overriding a prop or ?

Reply:
This is the only way I found to override event.stopPropagation in order to assert it was called